pandas/core/algorithms.py

```diff
@@ -692,30 +692,34 @@ def factorize(values, sort=False, order=None, na_sentinel=-1, size_hint=None):
 
 
 def value_counts(
-    values, sort=True, ascending=False, normalize=False, bins=None, dropna=True
+    values,
```

Comment:
Can type this as ndarray I think

Reply:
the `name = getattr(values, "name", None)` suggests that it could be a Series or Index

Reply:
Do you think `AnyArrayLike` from pandas._typing is applicable here?

Reply:
im hoping to refactor such that this function (as many as possible in this module, really) doesn't handle Series or Index so prefer not to make this official

Reply:
Sounds good